src/app-logic/url-handling.js

```diff
@@ -1179,6 +1179,51 @@ const _upgraders: {|
       query.transforms = transformStrings.map(upgradeTransformString).join('~');
     }
   },
+  [10]: ({ query }: ProcessedLocationBeforeUpgrade, profile?: Profile) => {
+    // This version changes the "collapse resource" transform to have a different
+    // collapsedFuncIndex: In version 9, "collapsed resource" functions were
+    // added to the funcTable by the transform as needed. In version 10, the
+    // "collapsed resource" functions are reserved at the beginning of the
+    // profile processing pipeline and get the following funcIndex:
+    // funcTable.length + resourceIndex.
+    // This was done so that funcIndexes are more stable and can't be affected
+    // by the transform stack.
+    //
+    // The "collapse resource" transform is not used a lot, and there are
+    // probably not a lot of URLs with it in circulation, so we just do the bare
+    // minimum here: If a single thread is selected, and we have access to the
+    // funcTable for that thread, then we compute the correct collapsed funcIndex.
+    // Otherwise we don't bother and leave incorrect funcIndexes in the URL.
+    // These funcIndexes will match *some* reserved resource func, so the UI
+    // shouldn't break in that case.
+
+    if (!query.transforms || !query.thread || !profile) {
+      return;
+    }
+
+    const selectedThreads = decodeUintArrayFromUrlComponent(query.thread);
+    if (selectedThreads.length !== 1) {
+      return;
+    }
+
+    const threadIndex = selectedThreads[0];
+    const funcTableLength = profile.threads[threadIndex].funcTable.length;
+
+    //     cr-{implementation}-{resourceIndex}-{wrongFuncIndex}
+    //  -> cr-{implementation}-{resourceIndex}-{correctFuncIndex}
+    function upgradeTransformString(transformString) {
```

Comment:
My understanding is that we're not changing any other transform than the "collapse resource" one, but the ones that work on func indexes (eg: drop-function, focus-function...) or call node path (eg: focus-subtree) are also affected. I think there are very few URLs that would have both collapse-resource and then one of those (though collapsing then focusing isn't unreasonable), so I believe it's OK to not handle this, but it would be good to make this choice explicit in a comment.

Reply:
You're totally right, I had forgotten about this. I'll add a comment.

Reply:
Comment added.